### DISCUSSION TITLE
Seek Bar Colors?

### DISCUSSION THREAD

**Discussion by superheroism** [tomasklaen/uosc#819] [Q&A] [answered]
Hello!  In the preview on the UOSC GitHub, the seek bar appears to shows colors and patterns depending on chapter and highlights the played duration blue.

![281727148-9f99f2ae-3b65-4935-8af3-8b80c605f022](https://github.com/tomasklaen/uosc/assets/58414264/b78b48cb-ae4e-4e68-86e1-84ad729b70d3)

I ran the installer and it appears to be working properly, but my entire seek bar is black regardless of time played or chapters discovered:

![image](https://github.com/tomasklaen/uosc/assets/58414264/d0c0de31-a79b-442e-b44a-62bda0313934)

Is there a configuration line I need to change?

**Comment by po5** [comment]
Ctrl+F `chapter` in the sample config file.
https://github.com/tomasklaen/uosc/blob/28e968450fb7c383353ab72722f39be4add329e0/src/uosc.conf#L203-L218

**Reply by superheroism** [reply]
I'm using the default configuration, so it's the same configuration file as the one you showed.  Like the one you linked, it assigns colors to openings, endings, and ads.  However, I still get the output shown in the original post.  Do I need to change something for the colors to appear?  It looks like the lines assigning colors (chapter_ranges, chapter_range_patterns) should execute, so I am expecting colors but see none.

**Reply by po5** [reply, marked as answer]
It seems from your screenshot that your chapters do not have titles. But I can't know for sure. You have provided zero information.

**Reply by superheroism** [reply]
I will try to find some content that follows a different title structure.  It was not clear to me that proper labeling was required to get this feature to work; I thought perhaps it was some internal algorithm, as is present in some other media players.  Thanks, though!

**Reply by tomasklaen** [reply]
We just match chapter titles, looking for strings like op, ed, opening, ending, intro, outro, and sponsorblock signature titles and IDs.

If your video file has untitled chapters, there won't be any colored chapter ranges in the timeline, as there's no reliable way for us to identify them.

**Reply by kenarsene** [reply]
> We just match chapter titles, looking for strings like op, ed, opening, ending, intro, outro, and sponsorblock signature titles and IDs.
> 
> If your video file has untitled chapters, there won't be any colored chapter ranges in the timeline, as there's no reliable way for us to identify them.


@tomasklaen 
I have similar issue, specifically the "intros/outros" never get colored, the openings/endings works fine though, also if I add "Intro/s , intro/s" to the 'openings' custom pattern it does get colored.

**Reply by tomasklaen** [reply]
You can see the current matching patterns here: https://github.com/tomasklaen/uosc/blob/6fa34c31d0a5290dee83282205768d15111df7d8/src/uosc/lib/utils.lua#L651-L654

If you have an idea how to improve them, PR welcome.
